src/main/java/seedu/address/model/task/UniqueTaskList.java

```diff
@@ -61,6 +61,26 @@ public void add(Task toAdd) throws DuplicateTaskException {
     }
 
     /**
+     * Updates a task to the list.
+     * 
+     * @throws DuplicateTaskException if the task to add is a duplicate of an existing task in the list.
+     */
+    public void update(ReadOnlyTask toUpdate, Task updatedTask) throws DuplicateTaskException {
```

Comment:
Because when you get a task from the UniqueTaskList, it's of the type ReadOnlyTask.